bottle.py

```diff
@@ -842,11 +842,15 @@ def default_error_handler(self, res):
 
     def _handle(self, environ):
         path = environ['bottle.raw_path'] = environ['PATH_INFO']
+        
         if py3k:
             try:
                 environ['PATH_INFO'] = path.encode('latin1').decode('utf8')
-            except UnicodeError:
-                return HTTPError(400, 'Invalid path string. Expected UTF-8')
+            except UnicodeError: 
+                try: # BUG 602 encoding path to latin1 may contain char not in utf8
+                    environ['PATH_INFO'] = urlunquote(path) 
```

Comment:
This does not seem safe to me.  Repeated UTF-8 decoding might succeed (by accident), mangling actual data.

Besides, the WSGI spec requires environ['PATH_INFO'] to be raw bytes from the HTTP request, decoded as Latin-1.  If you mount a WSGI app, bottle should convert it back before passing it to the mounted app.

IOW I think it would be better to fix this in `mount()`.